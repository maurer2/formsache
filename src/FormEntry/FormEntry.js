import React from 'react';
import PropTypes from 'prop-types';
import styles from './FormEntry.module.scss';

const FormEntry = ({label, name, value, type, handleChange, isValid})=> (
  <div className={`${styles.formEntry} ${isValid ? '' : styles.formEntryIsInvalid} `}>
    <label className={styles.formLabel} htmlFor={name}>
      {label}
    </label>
    <input
      className={styles.formField}
      value={value}
      onChange={handleChange}
      type={type}
      name={name}
      id={name}
    />
  </div>
);

export default FormEntry;

FormEntry.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  isValid: PropTypes.bool.isRequired,
};
