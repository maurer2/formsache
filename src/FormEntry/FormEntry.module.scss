.formEntry {
  display: flex;
  margin-bottom: 2rem;
  flex-direction: column;
}

.formLabel {
  display: flex;
  margin-bottom: 0.75rem;
}

.formField {
  padding: 0.75rem 1rem;
  width: 100%;
  min-width: 0;
  border: 2px solid black;
  outline-offset: -1px;
}

.formEntryIsInvalid {
  color: #de313f;

  .formLabel:after {
    content: '\00d7';
    margin-left: auto;
    font-size: 1.15rem;
    font-weight: bold;
    line-height: 1;
  }

  .formField {
    border-color: red;
  }
}