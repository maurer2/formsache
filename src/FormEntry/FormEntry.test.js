import React from 'react';
import { shallow } from 'enzyme';
import FormEntry from './FormEntry';

describe('FormEntry', () => {
  let wrapper;

  const mockedChangeFunction = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<FormEntry
      label="Cat"
      name="cat"
      value="purrfect"
      type="text"
      placeholder="Sir Meows a lot"
      handleChange={mockedChangeFunction}
      isValid
    />);
  });

  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
