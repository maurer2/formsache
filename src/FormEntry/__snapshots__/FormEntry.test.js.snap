// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormEntry snapshot 1`] = `
<div
  className="formEntry  "
>
  <label
    className="formLabel"
    htmlFor="cat"
  >
    Cat
  </label>
  <input
    className="formField"
    id="cat"
    name="cat"
    onChange={[MockFunction]}
    placeholder="Sir Meows a lot"
    type="text"
    value="purrfect"
  />
</div>
`;
